core/server/models/base/index.js

```diff
@@ -763,7 +775,16 @@ ghostBookshelf.Model = ghostBookshelf.Model.extend({
         const id = options.id;
         const model = this.forge({id: id});
 
-        data = this.filterData(data);
+        data = this.filterData(data, {
+            methodName: 'edit',
+            context: options.context
+        });
+
+        if (options.context && !options.context.internal && data.id && data.id !== options.id) {
+            return Promise.reject(new common.errors.ValidationError({
+                message: common.i18n.t('errors.models.base.editId')
+            }));
+        }
```

Comment:
We filter the data now, but we keep the `data.id` 🤔 - or?

If we would switch this condition with `filterData` - we can first validate the id and then remove it from the data object (if !internal).

What was your thinking here? Maybe i oversee something, i have not checked out your PR.

Reply:
Yeah we keep `data.id` because if we filter it out then we can't know if it was different to the `options.id` and so won't know if we should error. I figured we should run `filterData` before dealing with any properties on the data.

Are you saying we should remove the `id` prop in `filterData` and run that after the check for the error?

Reply:
I updated to what I think you mean :+1:

Reply:
Yeah this looks more correct 👍 Thanks